lib/Optimizer/Dialect/Quake/QuakeOps.cpp

```diff
@@ -54,6 +54,55 @@ LogicalResult quake::verifyWireArityAndCoarity(Operation *op) {
   return op->emitOpError("arity does not equal coarity of wires");
 }
 
+bool quake::isSupportedMappingOperation(Operation *op) {
+  return isa<OperatorInterface, MeasurementInterface, SinkOp>(op);
+}
+
+mlir::ValueRange quake::getQuantumTypesFromRange(mlir::ValueRange range) {
+
+  // Skip over classical types at the beginning
+  int numClassical = 0;
+  for (auto operand : range) {
+    if (!isa<RefType, VeqType, WireType>(operand.getType()))
+      numClassical++;
+    else
+      break;
+  }
```

Comment:
This works for all our current quake ops, of course. We'd have to revisit this if we ever have an op that interleaves classical data types with quantum data types. I think that's ok though, since we're checking the operation's interface as well.